src/Microsoft.ML.Data/Evaluators/AnomalyDetectionEvaluator.cs

```diff
@@ -665,12 +665,12 @@ private protected override void PrintFoldResultsCore(IChannel ch, Dictionary<str
 
             IDataView overall;
             if (!metrics.TryGetValue(MetricKinds.OverallMetrics, out overall))
-                throw Host.Except("No overall metrics found");
+                throw Host.ExceptSchemaMismatch(nameof(overall.Schema), "overall metrics", MetricKinds.OverallMetrics);
 
             // Find the number of anomalies, and the thresholds.
             int numAnomIndex;
             if (!overall.Schema.TryGetColumnIndex(AnomalyDetectionEvaluator.OverallMetrics.NumAnomalies, out numAnomIndex))
-                throw Host.Except("Could not find the 'NumAnomalies' column");
+                throw Host.ExceptSchemaMismatch(nameof(overall.Schema), "number of anomalies", AnomalyDetectionEvaluator.OverallMetrics.NumAnomalies);
```

Comment:
>number [](start = 73, length = 6)

I would not change the name of the column, in fact, I would just use the constant AnomalyDetectionEvaluator.OverallMetrics.NumAnomalies.